app/core/voyagerUtils.js

```diff
@@ -0,0 +1,93 @@
+
+module.exports = {
+
+  /**
+   * Creates a level status map using the list of level session objects.
+   * @param {Object[]} sessions - The list of level session objects.
+   * @returns {Object} - Object with key as the level original id, and value as complete/started.
+   */
+  getLevelStatusMap: function (sessions) {
+    let levelStatusMap = {}
+    for (let session of sessions) {
+      let levelOriginal = (session.level || session.get('level') || {}).original
+      if (levelOriginal && levelStatusMap[levelOriginal] !== 'complete') { // Don't overwrite a complete session with an incomplete one
+        if ((session.state || session.get('state') || {}).complete) {
+          levelStatusMap[levelOriginal] = 'complete'
+        } else {
+          levelStatusMap[levelOriginal] = 'started'
+        }
+      }
+    }
+    return levelStatusMap
+  },
+
+  /**
+   * Creates a level data map using the list of level objects.
+   * @param {Object[]} levels - The list of level objects.
+   * @returns {Object} - Object with key as the level original id, and value as the object containing level's data.
+   */
+  getLevelDataMap: function (levels) {
+    let levelDataMap = {}
+    for (let level of levels) {
+      let levelOriginal = level.original || level.get('original')
+      if (levelOriginal) {
+        levelDataMap[levelOriginal] = level.attributes || level
+      }
+    }
+    return levelDataMap
+  },
+
+  /**
+   * Gets the next level original id for a given level
+   * @param {Object} level - The level object.
+   * @param {Object[]} level.nextLevels - The array of nextLevels for the given level.
+   * @returns {string|undefined} - 'Original' id of the next level, or undefined.
+   */
+  getNextLevelOriginalForLevel: function (level) {
+    let nextLevels = level.nextLevels
+    if ((nextLevels || []).length > 0) {
+      return nextLevels[0].levelOriginal // assuming that there will be just one next level for voyager v1 as of now.
+    }
+    return undefined
+  },
+
+  /**
+   * Calculates all the next levels for a list of levels in a classroom/campaign and level sessions.
+   * @param {Object[]} sessions - The list of level session objects.
+   * @param {Object[]|Object} levels - The list of level objects, or an object with keys as level original id and value as level data.
+   * @param {Object[]} levels.nextLevels - The array of nextLevels for a level.
```

Comment:
Minor JsDoc nitpick. `@param` signifies a function argument.
If you want to describe the type of `Object` for `levels`, you can use [`@typedef`](http://usejsdoc.org/tags-typedef.html).

Reply:
I referred to this section(http://usejsdoc.org/tags-param.html#parameters-with-properties) for documenting some properties that are expected in the `levels object`.
I am not sure if we should use typedef here since it is about documenting `custom type`.

Reply:
I'm completely wrong! This solves a major pain point for me in my documenting!!!
Thanks for the reference!